src/api/dynamic_data.c

```diff
@@ -130,35 +156,39 @@ rosidl_dynamic_typesupport_dynamic_data_get_array_index(
 {
   RCUTILS_CHECK_ARGUMENT_FOR_NULL(dynamic_data, RCUTILS_RET_INVALID_ARGUMENT);
   RCUTILS_CHECK_ARGUMENT_FOR_NULL(array_index, RCUTILS_RET_INVALID_ARGUMENT);
-  return (dynamic_data->serialization_support->methods->dynamic_data_get_array_index)(
-    dynamic_data->serialization_support->impl, dynamic_data->impl, index, array_index);
+  return (dynamic_data->serialization_support->methods.dynamic_data_get_array_index)(
+    &dynamic_data->serialization_support->impl, &dynamic_data->impl, index, array_index);
 }
 
 
 rcutils_ret_t
 rosidl_dynamic_typesupport_dynamic_data_loan_value(
   rosidl_dynamic_typesupport_dynamic_data_t * dynamic_data,
   rosidl_dynamic_typesupport_member_id_t id,
-  rosidl_dynamic_typesupport_dynamic_data_t ** loaned_dynamic_data)
+  rcutils_allocator_t * allocator,
+  rosidl_dynamic_typesupport_dynamic_data_t * loaned_dynamic_data)
 {
   RCUTILS_CHECK_ARGUMENT_FOR_NULL(dynamic_data, RCUTILS_RET_INVALID_ARGUMENT);
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(allocator, RCUTILS_RET_INVALID_ARGUMENT);
```

Comment:
I think you need to check `rcutils_allocator_is_valid(allocator)` as part of argument checking (for all) - allocator could be non-null but contain null function pointers.

Reply:
I might defer this so we don't invalidate the windows debug CI run :x